inference/core/workflows/core_steps/transformations/dynamic_zones/v1.py

```diff
@@ -77,24 +85,42 @@ def get_execution_engine_compatibility(cls) -> Optional[str]:
         return ">=1.0.0,<2.0.0"
 
 
+def calculate_minimum_bounding_rectangle(
+    mask: np.ndarray,
+) -> Tuple[np.array, float, float, float]:
+    contours = sv.mask_to_polygons(mask)
+    largest_contour = max(contours, key=len)
+
+    rect = cv.minAreaRect(largest_contour)
+    box = cv.boxPoints(rect)
+    box = np.int0(box)
+
+    width, height = rect[1]
+    angle = rect[2]
+    return box, width, height, angle
+
+
 def calculate_simplified_polygon(
-    mask: np.ndarray, required_number_of_vertices: int, max_steps: int = 1000
-) -> np.array:
+    mask: np.ndarray,
+    required_number_of_vertices: int,
+    force_rectangle: bool = False,
+    max_steps: int = 1000,
+) -> Union[np.array, Tuple[np.array, float, float, float]]:
     contours = sv.mask_to_polygons(mask)
     largest_contour = max(contours, key=len)
 
-    # https://docs.opencv.org/4.x/d3/dc0/group__imgproc__shape.html#ga014b28e56cb8854c0de4a211cb2be656
+    if force_rectangle and required_number_of_vertices == 4:
```

Comment:
I'd move this to the `run` function and keep all logic relevant to forced rectangle together